Aquarius/Src/Aquarius/Events/EventHandler.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "Event.h"
+
+#include <GLFW/glfw3.h>
+#include <map>
+#include <vector>
+
+
+namespace Aquarius {
+
+    template <typename T>
+    class EventHandler
+    {
+    private:
+        using functionType = std::function<void(const Event<T>&)>;
+        std::map<T, std::vector<functionType>> subscribers;
+
+    public:
+        void subscribe(T Type, const functionType& function)
+        {
+            subscribers[Type].push_back(function);
```

Comment:
For subscribers where the event will change the state of the object, we will wanna make sure to use std::bind, and bind member functions and a respective instantiated object here

Reply:
Do you mean to use std::bind here or when we invoke it? I think you mean when we invoke it based on reading this [](https://en.cppreference.com/w/cpp/utility/functional/bind )

Reply:
When we invoke it yeah, this was just me thinking out loud haha